### PR TITLE
Piranhapoly

### PR DESCRIPTION
**Pull request by srajangarg**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -109,8 +109,12 @@ elseif ("${INTEGER_CLASS}" MATCHES "flint")
     set(SYMENGINE_INTEGER_CLASS "FLINT")
     set(WITH_FLINT yes)
 elseif ("${INTEGER_CLASS}" MATCHES "gmpxx")
-    set(SYMENGINE_INTEGER_CLASS "GMPXX")
-    set(WITH_GMPXX yes)
+    if (WITH_PIRANHA)
+        message(FATAL_ERROR "SymEngine can't be built with both INTEGER_CLASS=gmpxx and WITH_PIRANHA=yes")
```
Comment (isuruf) [inline CMakeLists.txt]: Add a warning here and set `set(SYMENGINE_INTEGER_CLASS "GMP")`

Comment (srajangarg) [inline CMakeLists.txt]: Instead of erroring, you mean?

Comment (isuruf) [inline CMakeLists.txt]: yes

**Comment by srajangarg** [comment]
@isuruf any changes apart from tests required?

**Comment by srajangarg** [comment]
Weird, the build with the same cmake settings is passing on my machine. https://travis-ci.org/symengine/symengine/jobs/136362214. Can someone confirm?

**Comment by srajangarg** [comment]
@isuruf ping

File: symengine/polys/uintpoly_piranha.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+integer_class UIntPolyPiranha::get_coeff(unsigned int x) const
+{
+    return poly_.find_cf(pmonomial{x});
+}
+
+integer_class UIntPolyPiranha::eval(const integer_class &x) const
+{
+    return piranha::math::evaluate<integer_class>(poly_,
```
Comment (isuruf) [inline symengine/polys/uintpoly_piranha.cpp]: Can you add a test for this?

**Pull request merged**